wallet/lib/pages/detailed_asset_view/widgets/nft_3d_asset.dart

```diff
@@ -1,31 +1,63 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
 import 'package:model_viewer_plus/model_viewer_plus.dart';
+import 'package:pylons_wallet/utils/constants.dart';
+import 'package:pylons_wallet/utils/image_util.dart';
+import 'package:webview_flutter/webview_flutter.dart';
 
 class Nft3dWidget extends StatefulWidget {
   final String url;
   final bool cameraControls;
   final Color backgroundColor;
+  final bool showLoader;
 
-  const Nft3dWidget(
-      {Key? key,
-      required this.backgroundColor,
-      required this.url,
-      required this.cameraControls})
-      : super(key: key);
+  const Nft3dWidget({Key? key, required this.backgroundColor, required this.url, required this.cameraControls, this.showLoader = false}) : super(key: key);
 
   @override
   _Nft3dWidgetState createState() => _Nft3dWidgetState();
 }
 
 class _Nft3dWidgetState extends State<Nft3dWidget> {
+  bool loading = true;
+
   @override
   Widget build(BuildContext context) {
-    return ModelViewer(
-      backgroundColor: widget.backgroundColor,
-      src: widget.url,
-      ar: false,
-      autoRotate: false,
-      cameraControls: widget.cameraControls,
+    return Stack(
+      children: [
+        ModelViewer(
+          backgroundColor: widget.backgroundColor,
+          src: widget.url,
+          ar: false,
+          autoRotate: false,
+          cameraControls: widget.cameraControls,
+          relatedJs: "",
+          javascriptChannel: {
+            JavascriptChannel(name: kLoadKey, onMessageReceived: (JavascriptMessage message) {}),
```

Comment:
Why we are not using these twi error and load channel?

Reply:
The issue mentioned was about loading the 3D model, I don't know the scenario of on Error in the 3D model case that is why It is there but not implemented yet. as far as the Load channel, we are doing the same thing with the Progress channel so I am going to remove Load channel

Reply:
we are loading in javascript any error can happen.